### PR TITLE
[AIRFLOW-XXX] Add tips for writing a note in UPDATIND.md

### PR DESCRIPTION
**Pull request by mik-laj**
Hi.

I think writing notes is a problem for developers, and it's not always helpful enough for end-users. I wrote a few short points that can improve future notes. This is a very important document when it comes to experience when migrating to the new version, so we should ensure that users enjoy reading it.

> - [ ] Previous behaviors

A good description of the previous behavior allows you to check if the problem is specific to you

> - [ ] New behaviors

It allows you to understand what has changed.

> - [ ] If possible, a simple example of how to migrate. This may include a simple code example.

People love examples. It's the easiest way to understand what you have to do.

> - [ ] If possible, the benefit for the user after migration e.g. "we want to make these changes to unify class names."

People don't like change. If they get the benefit they are willing to make changes.

> - [ ] If possible, the reason for the change, which adds more context to that interested, e.g. reference for Airflow Improvment Proposal.

A short note does not always allow you to understand the whole change. A reference to another document will help you better understand the change and will also involve new people in the community.

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: UPDATING.md [hunk 1]
```diff
@@ -41,6 +41,21 @@ assists users migrating to a new version.
 
 ## Airflow Master
 
+<!--
+
+I'm glad you want to write a new note. Make sure it contains the following information:
```
Comment (turbaszek) [inline UPDATING.md]:
```suggestion
I'm glad you want to write a new note. Remember that this note is intended for users. 
Make sure it contains the following information:
```
I would emphasise that those notes are mostly for users.

**Review by turbaszek — APPROVED** [review]

**Comment by potiuk** [comment]
I think it is far too many details at this stage (especially for 2.0 where we accumulate changes and there might contradicting or overlapping changes to do. We might revert some changes after we test it or we might want to combine some of the changes in which case there will be completely different migration procedure. 

I believe we should add all the details (especially migration procedure) before releasing 2.0.0 - when we will be writing the migration tool described in https://issues.apache.org/jira/browse/AIRFLOW-6390. 

This will be the right time to provide examples and test the migration process I think. 

Also we have the past entries which do not have all the details anyway, so we will have to do the exercise anyway.

For me current level of details we have in UPDATING.md was enough for now..

**Comment by mik-laj** [comment]
I think that the migration procedure will be best developed by the person who creates the change. She/He is aware of all the consequences. It may be very difficult for someone to understand all the edge cases. A good description will allow us to write the tool easier.  If we also have detailed descriptions, it will be easier for us to combine entries. It's best to start creating the migration procedure now. In the future it will be easier for us to improve than to build it from scratch.

**Comment by turbaszek** [comment]
> I think that the migration procedure will be best developed by the person who creates the change. She/He is aware of all the consequences. It may be very difficult for someone to understand all the edge cases.

This assumes that all authors are still active contributors and remember everything about the change. This is a really strong assumption imho. But that's where a good description should help.

**Comment by mik-laj** [comment]
This is the reason why I want to create a migration procedure now.  Other people may stop contributing to the project and creating this procedure will be difficult. Existing entries are another problem that requires a different solution.

**Comment by potiuk** [comment]
Ok. Let's try it - no harm in that. I already updated the description in #6950. Let's see if others will follow. 

What do you think we should do with the entries that have too short description from the past @mik-laj ? Do you want to ask authors of those changes to update the descriptions or you want to update it now ? or wait for release and update it there? 

IMHO - If you look at your point above:
> This is the reason why I want to create a migration procedure now. Other people may stop contributing to the project and creating this procedure will be difficult. Existing entries are another problem that requires a different solution.

the natural consequence is to follow up with the authors and ask them to provide missing information. Does it make sense? Are you willing to do that @mik-laj ?

**Pull request merged**